protein_lm/modeling/models/apt/model_pytorch.py

```diff
@@ -593,6 +704,10 @@ class APTLMHeadModel(GPT2PreTrainedModel):
     def __init__(self, config):
         super().__init__(config)
         self.transformer = APTModel(config)
+
+        #muP TO DO: check proper behavior for LM head, nothing should be done (?)
+        #see https://github.com/cofe-ai/Mu-scaling/blob/a3d13c3eaa02e28cc6ef95116358b128424f9fe4/modeling/modeling_gpt2_mup.py#L472
+        #see also table 8's caption in https://arxiv.org/pdf/2203.03466.pdf
         self.lm_head = nn.Linear(config.n_embd, config.vocab_size, bias=False)
```

Comment:
OK, I've looked into it and agree that we need to switch from the `nn.Linear` layer to `MuReadout`. Do we plan to use weight tying? Because `mup` also has a `MuSharedReadout` (https://github.com/microsoft/mup/blob/19814971934ef91dd546f88e913fc963e096d11c/mup/layer.py#L59-L68) that is likely handy in this case. For now I'll add a note about weight tying and just switch to the normal `MuReadout`.

Reply:
Thinking about it, we should probably just add a config option about weight tying, and support both cases accordingly.

Reply:
Actually, the [MuScaling repo adds](https://github.com/cofe-ai/Mu-scaling/blob/a3d13c3eaa02e28cc6ef95116358b128424f9fe4/modeling/utils.py#L15) a `width_mult` config option to `MuReadout `because the original `mup` repo has it [calculated and set on a layer by layer basis](https://github.com/microsoft/mup/blob/19814971934ef91dd546f88e913fc963e096d11c/mup/infshape.py#L22). I think we might want to follow the MuScaling approach of just integrating MuReadout and MuSharedReadout in our code accordingly.

Reply:
Sorry, I'm not quite sure I'm understanding -- are you suggesting that we write our own MuReadout and MuSharedReadout modules ourselves?